### PR TITLE
[Contribution] + fix tree

### PR DESCRIPTION
**Pull request by fkroockmann**
- Fix manage of limit for get the next result
- Delete a "next result" button after load the page
- Change the tree action to a service
- update TreeView (Export html to a file and add event for the action of tree)

### CODE REVIEW

File: src/tb/component/treeview/PopinTreeView.js [hunk 1]
```diff
@@ -1,75 +1,133 @@
...
+        'use strict';
+
+        var PopInTreeview = new JS.Class({
+
+                mainSelector: Core.get('wrapper_toolbar_selector'),
+
+                defaultConfig: {
+                    height: 300,
+                    width: 350,
+                    title: "Page tree",
```
Comment (mickaelandrieu) [inline src/tb/component/treeview/PopinTreeView.js]: Did you miss translation ?

File: src/tb/component/treeview/PopinTreeView.js [hunk 2]
```diff
@@ -1,75 +1,133 @@
...
+                    this.isLoaded = false;
+
+                    this.treeView = TreeViewMng.createTreeView(null, this.options);
+
+                    this.checkParameters();
+
+                    this.popinTemplate = jQuery(Renderer.render(popinTemplate)).clone();
+                    this.popIn = this.createPopIn();
+                    this.popIn.setContent(this.popinTemplate);
+                    this.popIn.addOption("create", jQuery.proxy(this.initOnCreate, this));
```
Comment (mickaelandrieu) [inline src/tb/component/treeview/PopinTreeView.js]: Same here

File: src/tb/component/treeview/PopinTreeView.js [hunk 3]
```diff
@@ -1,75 +1,133 @@
...
+                    this.popIn.addOption("open", jQuery.proxy(this.initOnOpen, this));
```
Comment (mickaelandrieu) [inline src/tb/component/treeview/PopinTreeView.js]: Same here

File: src/tb/component/treeview/templates/popin.twig [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<div class='bb5-windowpane-tree'>
+    <div class='action-ctn'>
+        <input class="show_folder_action" type='checkbox'/>{{ trans('show_folder') }}
+    </div>
+    <div class='bb5-treeview'>{{ trans('loading') }}...</div>
```
Comment (mickaelandrieu) [inline src/tb/component/treeview/templates/popin.twig]: :+1:

**Comment by mickaelandrieu** [comment]
Thanks, can you finish to translate ?

Good job.

**Comment by ndufreche** [comment]
@mickaelandrieu there is method actually to use the translator in javascript file instead of templates ?

**Comment by mickaelandrieu** [comment]
@ndufreche hmm, you can translate everywhere you can access to the component *but* you also need the configuration of the component.

Not so smooth, but possible.

**Comment by fkroockmann** [comment]
@mickaelandrieu, the configuration of the component is in all applications

**Comment by ndufreche** [comment]
Ok @fkroockmann will do the inner script translation in second time. Rebase your branch and will merge it.
:+1:

File: src/tb/component/treeview/PopinTreeView.js (hunk 2 above)
Comment (fkroockmann) [inline src/tb/component/treeview/PopinTreeView.js]: It's a function

File: src/tb/component/treeview/PopinTreeView.js (hunk 3 above)
Comment (fkroockmann) [inline src/tb/component/treeview/PopinTreeView.js]: It's function

**Comment by ndufreche** [comment]
Thanks @fkroockmann

**Pull request merged**